### DISCUSSION TITLE
Vite - Unpoly 2.2

### DISCUSSION THREAD

**Discussion by flihub** [unpoly/unpoly#244] [Q&A] [answered]
Hi,

when compile js bundle with [vite](https://vitejs.dev/) (v2.4.4). unpoly give error.

![image](https://user-images.githubusercontent.com/16720735/128415996-9afcde1d-16a9-4d77-91b4-03a444c5efed.png)

default config compiled to vendor.js and main.js.

```json
build: {
    outDir: './dist/',
    assetsDir: '',
    emptyOutDir: true,
    rollupOptions: {
      input: 'src/main.js',
      output: {
        entryFileNames: `[name].js`,
        chunkFileNames: `[name].js`,
        assetFileNames: `[name].[ext]`,
        // manualChunks: null,
      }
    },
  },
```

when set manualChunks to null only main.js compiled and no error.

Vite or Unpoly issue ?

**Comment by menzhessarov** [comment]
What is the benefit of using **Vite** with Unpoly?

**Comment by adam12** [comment]
Looking at the source of Unpoly, `SPEED_CALIBRATION` is defined on `window`, which I think was an omission when moving from CoffeeScript to ES6.

There's a commit (1cc197678289e5babe1c1e99887de46a46cedfab) just a few minutes ago that changes it to be defined via `const`, which should bring it into the lexical scope of that file and less chance it's undefined through bundle splitting or tree shaking when Rollup runs.

So my suggestion is to try 2.2.1 when it's released and report back.

**Comment by triskweline** [comment]
2.2.1 is now released with the changes described by @adam12.

**Comment by flihub** [comment]
with v2.2.1 : errors changed:

![image](https://user-images.githubusercontent.com/16720735/128507537-27ee2550-e8b3-4150-8819-d0016c617bf7.png)

and another bug: 

vite gives: vendor main js module files: 
<script type='module' src='{$vendor}'></script>
<script type='module' src='{$main}'></script>

up.log.config.banner = false  // not working

when no vendor js script tag doesnt need module attr. banner = false works.
<script src='{$main}'></script>

**Reply by johndwells** [reply]
@flihub how are you importing unpoly within `src/main.js'`?

I'm using Vite (same versions as you) and do not see this error, unless I try to import the CDN version of unpoly.js. But if I've defined unpoly as a package dependency and import the copy inside the local `node_modules` folder, it builds fine.

**Reply by flihub** [reply]
![image](https://user-images.githubusercontent.com/16720735/129179708-3dc31375-e2c7-43a4-b2ae-9454440ab8cc.png)

**Reply by flihub** [reply]
tested 2.3.0 

![image](https://user-images.githubusercontent.com/16720735/131503002-dc70842e-7336-4071-9ec1-4bd45578d2e1.png)

**Reply by adam12** [reply]
@flihub can you make an example repo with Vite/Unpoly setup that replicates that error?

**Reply by flihub** [reply]
hi,

finally found what cause error.

"dist/vendor.js?v=" . filemtime("dist/vendor.js")

i was disabled vite version hash by:

entryFileNames: `[name].js`,
chunkFileNames: `[name].js`,
assetFileNames: `[name].[ext]`,

main.js has vendor.js but type module has ?v= 
import{U as e,I as t}from"./vendor.js"

https://stackoverflow.com/questions/45943972/how-to-handle-caching-of-javascript-files-imported-as-es6-module

**Comment by triskweline** [comment, marked as answer]
For some background on why this happens, please see [my comment](https://github.com/unpoly/unpoly/issues/136#issuecomment-897106764) in #136.

**Reply by flihub** [reply]
thanks.
